src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/UserService.java

```diff
@@ -26,37 +28,78 @@
     private final Logger log = LoggerFactory.getLogger(UserService.class);
 
     private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UniversityDomains universityDomains;
 
-    @Autowired public UserService(@Qualifier("userRepository") UserRepository userRepository) {
+    @Autowired
+    public UserService(@Qualifier("userRepository") UserRepository userRepository,
+        PasswordEncoder passwordEncoder, UniversityDomains universityDomains) {
         this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.universityDomains = universityDomains;
     }
 
     public List<User> getUsers() {
         return this.userRepository.findAll();
     }
 
-    public @NotNull User createUser(User newUser) {
+    public @NotNull User createUser(User newUser, User createdBy) {
+        log.debug("creating user {}", newUser);
+
+        newUser.setId(null);
         newUser.setCreationDate(new Date());
-        newUser.getAuthenticationData().setSalt(Utils.generateSalt(16));
```

Comment:
Shouldn't we use a static class variable here?     Utils.SALT_LENGTH

Reply:
as you can see, this was already removed and changed